### PR TITLE
ROX-11323 Enable fake workloads in local-sensor

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

These changes enable the fake workloads in `local-sensor`

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

```
go run tools/local-sensor/main.go -record -resync=0s -with-fakeworkload=scale/workloads/default.yaml
```

generates a trace file (`k8s-trace.jsonl`) with fake events.

### CODE REVIEW

File: sensor/kubernetes/fake/fake.go [hunk 1]
```diff
@@ -85,14 +85,33 @@ type WorkloadManager struct {
...
 }
 
+// WorkloadManagerConfig WorkloadManager's configuration
+type WorkloadManagerConfig struct {
+	workloadFile string
+	client       client.Interface
+}
+
+// ConfigDefaults default configuration
+func ConfigDefaults() *WorkloadManagerConfig {
```
Comment (vikin91) [inline sensor/kubernetes/fake/fake.go:95]: How about renaming this to `NoFakeWorklod()` to highlight that the default behavior is not to use any fake workload?

File: sensor/kubernetes/fake/fake.go [hunk 2]
```diff
@@ -85,14 +85,33 @@ type WorkloadManager struct {
 	networkManager      manager.Manager
 }
 
+// WorkloadManagerConfig WorkloadManager's configuration
+type WorkloadManagerConfig struct {
+	workloadFile string
+	client       client.Interface
```
Comment (vikin91) [inline sensor/kubernetes/fake/fake.go:91]: Why do we have client here when no methods use that field? This suggests that maybe it should be a var outside of this struct (or shall simply be reamoved)?

**Review by vikin91 — COMMENTED** [review]
Looks really cool!

Let's maybe clarify the `client       client.Interface` in the `WorkloadManagerConfig` before we merge.

File: sensor/kubernetes/fake/fake.go (hunk 1 above)
Comment (lvalerom) [reply sensor/kubernetes/fake/fake.go:95]: I see where you're coming from but I think that renaming this function to `NoFakeWorkload` could lead to a misunderstanding. `ConfigDefaults` takes the default path as `/var/scale/stackrox/workload.yaml` from `workloadPath`. The decision whether we use fake workloads or not is based on the existence of the configured file (whether it is the default file previously mentioned or a configured one using `WithWorkloadFile`)

File: sensor/kubernetes/fake/fake.go (hunk 2 above)
Comment (lvalerom) [reply sensor/kubernetes/fake/fake.go:91]: Oops that's left over from when I was spiking on this 😅 I'll remove it.

File: sensor/kubernetes/fake/fake.go (hunk 1 above)
Comment (vikin91) [reply sensor/kubernetes/fake/fake.go:94]: Ah, I see. Okay then 👍

**Review by vikin91 — APPROVED** [review]

File: tools/local-sensor/main.go [hunk 3]
```diff
@@ -191,11 +199,19 @@ func registerHostKillSignals(startTime time.Time, fakeCentral *centralDebug.Fake
 // If a KUBECONFIG file is provided, then local-sensor will use that file to connect to a remote cluster.
 func main() {
 	localConfig := mustGetCommandLineArgs()
+	var fakeClient client.Interface
 	fakeClient, err := k8s.MakeOutOfClusterClient()
 	// when replying a trace, there is no need to connect to K8s cluster
 	if localConfig.ReplayK8sEnabled {
 		fakeClient = k8s.MakeFakeClient()
 	}
+	var workloadManager *fake.WorkloadManager = nil
```
[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply tools/local-sensor/main.go:208]: > I wonder why it didn't in the pre-merge check? 🤔

Yeah, me too 🤔 Did anyone figure that out?

[the remark this answers is not in the document]
Comment (porridge) [reply tools/local-sensor/main.go:208]: My assumption was that the linter or its configuration was changed concurrently. I bet our lint master @janisz would know :-)

**Pull request merged**